### ISSUE TITLE
The result of adding the same variable formulas in the KAN Compiler is not ideal.（KAN  Compiler 对同一变量相加公式的结果不理想。）

### ISSUE THREAD

**Issue by VVeida** [KindXiaoming/pykan#451]
import time
import torch
import numpy as np
from kan import *
torch.set_default_dtype(torch.float64)

from kan.compiler import kanpiler
from sympy import *
from kan.utils import create_dataset
import torch
from kan.feynman import get_feynman_dataset
import matplotlib.pyplot as plt

def safemakefile(save_path):
    if not os.path.exists(save_path):
        os.makedirs(save_path)
def setup_seed(seed):
    np.random.seed(seed)
    random.seed(seed)
    
device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
print(device)
t = symbols('t')
input_variables = (t,)
f = cos(60*pi*t+10*sin(2*pi*t))
model  = kanpiler(input_variables, f).to(device)
f = lambda t: torch.cos(60*torch.pi*t+10*torch.sin(2*torch.pi*t))
dataset = create_dataset(f, n_var=1, device=device)
model.get_act(dataset)  
model.plot()
![1](https://github.com/user-attachments/assets/9d67891a-6c40-4f91-8a89-7b3ace476705)

**Comment by KindXiaoming** [comment]
I guess this is expected 
* (0,0,0) is $60\pi t+10{\rm sin}(2\pi t)$. This looks linear because 10 << 60.
* (1,0,0) is cos. This looks super oscillatory because the input range is very large so there are many cycles.
